instrumentation/action_mailer/lib/opentelemetry/instrumentation/action_mailer/instrumentation.rb

```diff
@@ -27,7 +27,9 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         end
 
         option :disallowed_notification_payload_keys, default: [], validate: :array
+        option :disallowed_process_payload_keys,      default: [], validate: :array
         option :notification_payload_transform,       default: nil, validate: :callable
+        option :process_payload_transform,            default: nil, validate: :callable
```

Comment:
I get what you're saying in your PR description about the benefits of having additional configuration options for the process notification. 

I think we could use some documentation that goes into a little more detail about the scope of these different configs. 

The ActionView instrumentation has similar configuration options, and documents them like this: https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/instrumentation/action_view/lib/opentelemetry/instrumentation/action_view/instrumentation.rb#L10-L45 

Would you mind adding something similar for ActionMailer?